pyuploadcare/resources/file.py

```diff
@@ -353,45 +369,229 @@ def construct_from(cls, file_info):
         file_._info_cache = file_info
         return file_
 
+    @staticmethod
+    def _get_file_size(file_object: IO) -> int:
+        return os.fstat(file_object.fileno()).st_size
+
     @classmethod
-    def upload(
+    def upload(  # noqa: C901
         cls,
-        file_obj,
+        file_obj_or_url: Union[IO, str],
```

Comment:
```suggestion
        file_handle: Union[IO, str],
```
is it a better name?

Reply:
done